### ISSUE TITLE
FocalLoss from_logits

### ISSUE THREAD

**Issue by xoiga123** [qubvel/segmentation_models.pytorch#612]
Why do `JaccardLoss`, `DiceLoss`, `TverskyLoss`, and `LovaszLoss` all have a parameter called `from_logits` to switch between logits and non-logits output, but `FocalLoss` doesn't? Apparently, it is hardcoded in the code as
```python
logpt = F.binary_cross_entropy_with_logits(output, target, reduction="none")
```
while it could be something like
```python
if self.from_logits:
    logpt = F.binary_cross_entropy_with_logits(output, target, reduction="none")
else:
    logpt = F.binary_cross_entropy(output, target, reduction="none")
```
Does it have something to do with the way you flatten the output in `forward`?
```python
if self.mode in {BINARY_MODE, MULTILABEL_MODE}:
    y_true = y_true.view(-1)
    y_pred = y_pred.view(-1)
    loss = self.focal_loss_fn(y_pred, y_true)
````
I also don't quite get why you flatten the output this way. Can you enlighten me, please?

And great repo btw, we've been using your loss functions in [our net](https://github.com/datvuthanh/HybridNets) :smile:
